src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_SIZE = 10;
+    public static final int DEFAULT_RADIUS = 10;
+    public static final String DEFAULT_COLOR = Color.WHITE.name();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Figures randomFigure = Figures.values()[random.nextInt(FIGURE_COUNT)];
+        switch (randomFigure) {
+            case SQUARE:
+                return getRandomSquare();
+            case RECTANGLE:
+                return getRandomRectangle();
+            case RIGHT_TRIANGLE:
+                return getRandomRightTriangle();
+            case CIRCLE:
+                return getRandomCircle();
+            case ISOSCELES_TRAPEZOID:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getRandomSquare() {
+        return new Circle(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE));

```

Comment:
You dont want to create Circle calling getSquare

Reply:
Sorry, thanks